### ISSUE TITLE
Data models with variadic functions cannot be API resources

### ISSUE THREAD

**Issue by aheuertz** [api-platform/core#354]
PHP 5.6 introduced [the '...' syntax for variadic functions](http://php.net/manual/en/migration56.new-features.php#migration56.new-features.variadics). When this is used in a resource class, a PHP parser (dependency) chokes.

Example variadic method in a resource class:

```
public function addModulePermission(ModulePermission ...$modulePermissions)
{
    foreach ($modulePermissions as $modulePermission) {
        $this->modulePermissions[] = $modulePermission;
    }

    return $this;
}
```

The error returned (shortened for brevity):

```
[Symfony\Component\Debug\Exception\ContextErrorException]  
Notice: Undefined offset: 389                              

Exception trace:
 () at /project/vendor/nikic/php-parser/lib/PHPParser/Lexer.php:121
 Symfony\Component\Debug\ErrorHandler->handleError() at /project/vendor/nikic/php-parser/lib/PHPParser/Lexer.php:121
 PHPParser_Lexer->getNextToken() at /project/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Lexer.php:48
 phpDocumentor\Reflection\Lexer->getNextToken() at /project/vendor/nikic/php-parser/lib/PHPParser/Parser.php:979
 PHPParser_Parser->parse() at /project/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Traverser.php:52
 phpDocumentor\Reflection\Traverser->traverse() at /project/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/FileReflector.php:163
 ...
```

I am using PHP's `func_get_args` temporarily to circumvent this.

**Comment by dunglas** [comment]
This a dependency problem:

* We use `phpdocumentor/reflection` 1.0
* `phpdocumentor/reflection` uses `nikic/php-parser` 0.9.4

And variadic functions are supported only by `nikic/php-parser` 1.x and 2.x. The PHPDocumentator team has released an alpha of the 2.0 version of `phpdocumentor/reflection` that uses `nikic/php-parser` 1.x. We will support variadic functions when the stable when the stable version will be released and we do the upgrade.

**Comment by Simperfit** [comment]
This should be resolved since we are using 3.0
